webextension/background.js

```diff
@@ -280,6 +280,20 @@ const messageHandler = {
       });
     });
 
+    browser.idle.onStateChanged.addListener((newState) => {
+      browser.tabs.query({}).then(tabs => {
+        for (let tab of tabs) { // eslint-disable-line prefer-const
+          if (newState === "idle" || newState === "locked") {
```

Comment:
Yes, `browser.idle.onStateChanged` seems to be a browser event - not a tab event. When the user goes idle, I send and reset all the tabs' "`per-activity`" page counters. The scenario here is:

1. User loads a few pages in a container tab
2. User switches to some other tab(s)
3. User goes idle

I'll see if I can move the `initTabCounter(tab, "user-became-active")` call into `browser.tabs.onActivated` so we only initialize an activity counter for the tab when the user activates it.

I'd be fine to drop `locked` state.